sliceutil/sliceutil_test.go

```diff
@@ -40,6 +40,35 @@ func TestJoin(t *testing.T) {
 	}
 }
 
+func TestJoinWith(t *testing.T) {
+	cases := []struct {
```

Comment:
Nice one @tigh-latte just one suggestion on this one, consider adding delimiter field so we could test out different delimiter options

Reply:
sweet done